### ISSUE TITLE
Can we add shap_values of different features?

### ISSUE THREAD

**Issue by ZhouYu44** [shap/shap#282]
This is not a bug or a real issue, but a general question regarding shap values. In our application, we have a lot of feature sets. For example, feature set A include features a1,a2,a3, feature set B include features b1, b2. Can we sum up the shap_values of a1,a2, and a3 and take that as the shap value for feature set A, and sum up shap_values of b1 and b2 and take that as shap_values for feature set B?

Using your example:

import xgboost
import shap
import pandas as pd
shap.initjs()
X,y = shap.datasets.boston()
model = xgboost.train({"learning_rate": 0.01}, xgboost.DMatrix(X, label=y), 100)
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X)

df_shap_values = pd.DataFrame(shap_values,columns =X.columns)

My question is can I sum up, for example, the first three columns of df_shap_values as the shap value 
for the feature set made of the first three columns? 

My naive understanding about shap_values tell me the answer seems to be NO. But wondering if there is any reasonable way that I could get this done? 

Thank you for looking into this!

**Comment by slundberg** [comment]
Good question. The summary is yes, you can sum Shapley values and SHAP values are just Shapley values of conditional expectations. So you can just add SHAP values and get the SHAP value of a group of features. 

A small caveat I should mention is that with TreeExplainer conditional dependence is estimated by conditioning on features higher in the tree (since that's the only dependence recorded in the passed tree model). Since this is only an approximation of the true conditional dependence under the data generating distribution it is not necessarily perfectly symmetric, and so I think you might get slightly different results summing features than if we were to hold them out as a group.

**Comment by ZhouYu44** [comment]
Got it. Very helpful. Thank you!!

**Comment by alexlaurence** [comment]
Any update on the type of analysis you did for grouped features?

**Comment by maxkitman** [comment]
> Good question. The summary is yes, you can sum Shapley values and SHAP values are just Shapley values of conditional expectations. So you can just add SHAP values and get the SHAP value of a group of features.

Hi @slundberg, thank you for your explanation. 

My question is then : should this be the sum of the original or absolute SHAP values from these different metrics ?
Let's assume that `met_1` and `met_2` represent the information from `met`. If `met_1` SHAP value is +5 and `met_2` SHAP value is -5, then should `met` SHAP value be 0 or 10 ?

My first impression is that the sum of the absolute values makes sense over rows to get features importance but I would not apply that over columns as in the previous example, I would not say that `met` has an effect on the target just because we could not define what is its effect (positive/negative).

Thank you for your answer.
